DTS-STN/secure-client-hub
# Add support for validateSession endpoint

## [ADO-93869](https://dev.azure.com/VP-BD/DECD/_workitems/edit/93869)

### Description
This PR adds the `ValidateSession` function within `auth.js` that sends a GET request to ECAS to continue the MSCA session. Currently we have to use a proxy for off network development so I've added a flag to make it possible to make this request locally off network. If not on proxy, the only way for this to work currently on any of our deployments is to access the site on-network (if auth is enabled, else it won't matter). ValidateSession will be called at most 1 time per 15 seconds on page clicks as the callback is throttled using lodash, and on each page load.

Fixes [AB#93869](https://dev.azure.com/VP-BD/DECD/_workitems/edit/93869)

### What to test for/How to test
1. Pull in branch & msg me for environment variables
2. Connect to the proxy
3. Type `npm run dev`
4. Navigate through the login/auth process
5. Once on the dashboard, click once and go back to your terminal. You should notice something that says `Session renewed!`
6. Navigate to the other pages and ensure you see the same message

### Additional Notes
If you don't have access to the proxy and would like to see a working example of this, reach out to me and I can demo for you.
